cmd/crc/cmd/setup.go

```diff
@@ -32,6 +32,16 @@ func runSetup(arguments []string) {
 	if err := validateSetupFlags(); err != nil {
 		errors.Exit(1)
 	}
+
+	// Proxy Setup
+	proxy, err := handleProxyConfig()
+	if err != nil {
+		errors.ExitWithMessage(1, err.Error())
+	}
+	if proxy != nil {
+		proxy.ApplyToEnvironment()
+	}
```

Comment:
Maybe this should be handled in the `download` package directly? Or are there more codepaths which need proxy?

Reply:
@cfergeau this will ensure that before running any code path, the proxy is apply to env, this doesn't hurt us and in future if we add any code path which require proxy shouldn't have to worry about adding it in that function.

Reply:
well... is might not hurt us, but it does have a side-effect, it applies changes to the users shell environment.